metricbeat/module/system/diskio/diskio.go

```diff
@@ -131,9 +139,27 @@ func (m *MetricSet) Fetch(r mb.ReporterV2) error {
 			MetricSetFields: event,
 		})
 		if !isOpen {
-			return nil
+			continue
```

Comment:
No reason to continue here, the module has been stopped, no more events will be published.
```suggestion
			return nil
```